pkg/controller/sshd/sshd_controller.go

```diff
@@ -0,0 +1,564 @@
+		return reconcile.Result{}, err
+	}
+	if !exists {
+		// It isn't bad that it doesn't exist if there's no error, so re-queue.
+		r.SetSSHDStatusPending(instance, reqLogger, "Waiting on service load balancers")
+		return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+
+	r.route53 = &Route53Data{
+		loadBalancerDNSName:      elb.DNSName,
+		loadBalancerHostedZoneId: elb.DNSZoneId,
+		resourceRecordSetName:    instance.Spec.DNSName + "." + clusterBaseDomain,
+		privateHostedZoneName:    clusterBaseDomain + ".",
+		// The public zone name omits the cluster name.
+		// e.g. mycluster.abcd.s1.openshift.com -> abcd.s1.openshift.com
+		publicHostedZoneName: clusterBaseDomain[strings.Index(clusterBaseDomain, ".")+1:] + ".",
+	}
+
+	err = r.ensureDNSRecords()
+	if err != nil {
+		r.SetSSHDStatusError(instance, reqLogger, "Failed to ensure the DNS record", err)
+		return reconcile.Result{}, err
+	}
+
+	r.SetSSHDStatus(instance, "SSHD is ready", cloudingressv1alpha1.SSHDStateReady)
+
+	return reconcile.Result{RequeueAfter: 60 * time.Second}, nil
+}
+
+func getMatchLabels(cr *cloudingressv1alpha1.SSHD) map[string]string {
+	return map[string]string{"deployment": cr.Name}
+}
+
+func newSSHDDeployment(cr *cloudingressv1alpha1.SSHD, configMapList *corev1.ConfigMapList) *appsv1.Deployment {
+	replicas := int32(1)
+	terminationGracePeriod := int64(30) // seconds
+	revisionHistoryLimit := int32(10)
+	progressDeadline := int32(600) // seconds
+
+	// Prefer nil over empty slices to satisfy reflect.DeepEqual.
+	var volumes []corev1.Volume
+	var volumeMounts []corev1.VolumeMount
+	volumeDefaultMode := int32(0600)
+
+	for _, configMap := range configMapList.Items {
+		volumeName := configMap.ObjectMeta.Name
+		volumes = append(volumes, corev1.Volume{
+			Name: volumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: configMap.ObjectMeta.Name,
+					},
+					DefaultMode: &volumeDefaultMode,
+				},
+			},
+		})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      volumeName,
+			MountPath: filepath.Join(authorizedKeysMountPath, volumeName),
+		})
+	}
+
+	sshdContainer := corev1.Container{
+		Name:    "sshd",
+		Image:   cr.Spec.Image,
+		Command: []string{"/opt/start-sshd.sh"},
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          "ssh",
+				ContainerPort: int32(2222),
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		VolumeMounts:             volumeMounts,
+		TerminationMessagePath:   "/dev/termination-log",
+		TerminationMessagePolicy: corev1.TerminationMessageReadFile,
+		ImagePullPolicy:          corev1.PullAlways,
+	}
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
```

Comment:
I think we will want to restrict the kinds of nodes on which this runs to at least infra and probably only on master nodes. Please add the necessary tolerations/nodeAffinity.

Reply:
+1 to this being a master workload

Reply:
@mbarnes Not seeing controls here to direct this workload on the master nodes? Missing the `nodeAffinity` portion of the spec?

Reply:
I used a `nodeSelector`.  Would `nodeAffinity` be preferred?

Reply:
Added a commit to use `nodeAffinity`.